### PR TITLE
[OD-619] Configure ckanext-twitter popup trigger

### PR DESCRIPTION
**Pull request by ankitbrana**
[[OD-619] Configure ckanext-twitter popup trigger](https://opengovinc.atlassian.net/browse/OD-619)

## Description
**Introduction**
This PR to update the trigger conditions for CHHS's twitter popup. It will now only pop up in the following scenarios:

Twitter popup is triggered in the following use cases:

CHHS would prefer to only have the Twitter popup triggered under the following cases:
1. When a dataset is created
2. When a resource within a twitter enabled dataset is added or edited

Not triggered when:
1. When a dataset is updated (metadata edits)

**Changes:**
- Updated plugin.py to implement ResourceController instead of PackageController for the after_update and after_create functions
- The after_create function in plugin.py was added so that after a resource is created it can be called to update the session to to be suitable for tweeting
- The send function was updated in tweet.py so that the session is properly updated to no longer be suitable for testing for the given package since the tweet will be sent
- Refactored the way formatstring was being initialized for clarity in parsers.py

## Test Procedure & Approval Criteria 
**Test Procedures:**
- Install the CHHS Schema Extension to your VM [here](https://github.com/OpenGov-OpenData/ckanext-chhs_schema) 
- Install the twitter extension to your VM from this fork on the chhs branch
- Add twitter credentials to your .ini if available (Test credentials for ckan_test twitter account found in CHHS sandbox's .ini)

Test Trigger Conditions

Tests with Twitter Popup Enabled

1. Create a new dataset w/ twitter popup in dataset enabled :
- Create a dataset with one resource and enable the twitter popup field
- Verify that the twitter popup appears with the tweet formatted indicating that it is a new dataset

2. Updated a resource w/ twitter popup in dataset enabled:
- Update the description of a resource in a dataset with twitter popup enabled
- Verify that a twitter popup should appear indicating that the dataset was updated

3. Verify twitter popup does not appear when editing dataset metadata w/ twitter popup in dataset enabled:
- Update a dataset's metadata that has the twitter popup enabled
- Verify that the twitter popup does not appear after editing the metadata

Tests with Twitter Popup Disabled

4. Create a new dataset w/ twitter popup disabled in dataset :
- Create a dataset with one resource and enable the twitter popup field
- Verify that the twitter popup should not appear

5. Updated a resource w/ twitter popup in dataset disabled:
- Update the description of a resource in a dataset with twitter popup enabled
- Verify that a twitter popup should not appear

6. Verify twitter popup does not appear when editing dataset metadata w/ twitter popup in dataset disabled:
- Update a dataset's metadata that has the twitter popup enabled
- Verify that the twitter popup does not appear after editing the metadata

## Submitter Checklist
- [x] I have checked the README file and updated it when necessary
- [x] I have tested the changes locally

## Reviewer Checklist

- [ ] I have run this code change locally
- [x] The description clearly states the impact of this code change.
- [x] There is enough information to verify the functionality

### CODE REVIEW

File: ckanext/twitter/plugin.py [hunk 1]
```diff
@@ -38,17 +34,31 @@ def update_config(self, config):
         # Add resources
         p.toolkit.add_resource('theme/fanstatic', 'ckanext-twitter')
 
-    # IResourceController
+
+    # IResourceController, called after a resource is created
+    def after_create(self, context, res_pkg_dict):
+        data_dict = {'id': res_pkg_dict['package_id']}
```
Comment (jguo144) [inline ckanext/twitter/plugin.py]: The variable `res_pkg_dict` should probably be `resource` to align with other CKAN code.

File: ckanext/twitter/plugin.py [hunk 2]
```diff
@@ -38,17 +34,31 @@ def update_config(self, config):
...
+
+
+    # IResourceController, called after a resource is updated
     def after_update(self, context, res_pkg_dict):
-        if ckanext.twitter.lib.config_helpers.tweet_trigger() == 'dataset':
-            pkg_dict = res_pkg_dict
-        else:
-            data_dict = {'id': res_pkg_dict['package_id']}
-            pkg_dict = tk.get_action('package_show')(context, data_dict)
+        data_dict = {'id': res_pkg_dict['package_id']}
```
Comment (jguo144) [inline ckanext/twitter/plugin.py]: The variable `res_pkg_dict` should probably be `resource` to align with other CKAN code.

File: ckanext/twitter/plugin.py [hunk 3]
```diff
@@ -78,4 +88,4 @@ def before_map(self, _map):
                      conditions = {
                          'method': ['POST']
                          })
-        return _map
+        return _map
```
Comment (jguo144) [inline ckanext/twitter/plugin.py]: Python's PEP8 coding standards specify that a newline should be at the end of files.

**Comment by jguo144** [comment]
Looks good to me.

**Comment by ankitbrana** [comment]
Updated README and fixed the new line missing at EOF for plugin.py

**Pull request merged**